src/phone-input/country-select-dropdown.js

```diff
@@ -22,87 +22,97 @@ import {getOverrides} from '../helpers/overrides.js';
-      mapIsoToLabel,
-      country,
-      overrides = {},
-    } = props;
-    const [Container, containerProps] = getOverrides(
-      overrides.CountrySelectDropdown,
-      DefaultContainer,
-    );
-    const [ListItem, listItemProps] = getOverrides(
-      overrides.CountrySelectDropdownListItem,
-      DefaultListItem,
-    );
-    const [FlagColumn, flagColumnProps] = getOverrides(
-      overrides.CountrySelectDropdownFlagColumn,
-      DefaultFlagColumn,
-    );
-    const [NameColumn, nameColumnProps] = getOverrides(
-      overrides.CountrySelectDropdownNameColumn,
-      DefaultNameColumn,
-    );
-    const [Dialcode, dialcodeProps] = getOverrides(
-      overrides.CountrySelectDropdownDialcodeColumn,
-      DefaultDialcodeColumn,
-    );
-    const scrollIndex = Math.min(
-      children.findIndex(opt => opt.props.item.id === country.id) + 5,
-      children.length - 1,
-    );
-    return (
-      <Container ref={ref} $height={maxDropdownHeight} {...containerProps}>
-        <AutoSizer>
-          {({height, width}) => {
-            return (
-              <List
-                role="listbox"
-                height={height}
-                width={width}
-                rowCount={children.length}
-                rowHeight={42}
-                scrollToIndex={scrollIndex}
-                rowRenderer={({index, key, style}) => {
-                  // resetMenu and getItemLabel should not end up on native html elements
-                  const {resetMenu, getItemLabel, ...rest} = children[
-                    index
-                  ].props;
-                  return (
-                    <ListItem
-                      key={key}
-                      style={style}
-                      {...rest}
-                      {...listItemProps}
-                      data-e2e="country-select-list-item"
-                      data-iso={children[index].props.item.id}
-                    >
-                      <FlagColumn {...flagColumnProps}>
-                        <StyledFlag iso={children[index].props.item.id} />
-                      </FlagColumn>
-                      <NameColumn {...nameColumnProps}>
-                        {mapIsoToLabel
-                          ? mapIsoToLabel(props.children[index].props.item.id)
-                          : children[index].props.item.label}
-                      </NameColumn>
-                      <Dialcode {...dialcodeProps}>
-                        {children[index].props.item.dialCode}
-                      </Dialcode>
-                    </ListItem>
-                  );
-                }}
-              />
-            );
-          }}
-        </AutoSizer>
-      </Container>
-    );
-  },
-);
+CountrySelectDropdown.defaultProps = {
+  maxDropdownHeight: defaultProps.maxDropdownHeight,
+  overrides: {},
+};
 
-export default CountrySelectDropdown;
+function CountrySelectDropdown(props: CountrySelectDropdownPropsT) {
```

Comment:
may want to inline the union of `PropsT & {containerRef: ref}` here since it's an encapsulated implementation detail. no-one should be calling this function without the forwardRef wrapping

Reply:
oh that's a great point